akka-actor-typed-tests/src/test/scala/akka/actor/typed/DeadLetterSpec.scala

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (C) 2017-2023 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.actor.typed
+
+import akka.actor.testkit.typed.scaladsl.LogCapturing
+import akka.actor.testkit.typed.scaladsl.ScalaTestWithActorTestKit
+import akka.actor.typed.scaladsl.AskPattern.Askable
+import akka.actor.typed.scaladsl.AskPattern.schedulerFromActorSystem
+import akka.actor.typed.scaladsl.Behaviors
+import akka.actor.typed.scaladsl.Behaviors._
+import akka.util.Timeout
+import org.scalatest.wordspec.AnyWordSpecLike
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import scala.concurrent.TimeoutException
+import scala.concurrent.duration._
+import scala.math.multiplyExact
+import scala.util.Failure
+import scala.util.Success
+
+sealed trait Command
+case class Multiply(a: Int, b: Int, forwardRef: ActorRef[WorkerCommand], replyTo: ActorRef[Int]) extends Command
+case class ReplyResult(num: Int, replyTo: ActorRef[Int]) extends Command
+case class Ignore() extends Command
+
+sealed trait WorkerCommand
+case class WorkerMultiply(a: Int, b: Int, replyTo: ActorRef[WorkerResult]) extends WorkerCommand
+case class WorkerResult(num: Int) extends WorkerCommand
+
+class DeadLetterSpec extends ScalaTestWithActorTestKit("""
+    akka.loglevel=DEBUG
+    akka.actor.debug.event-stream = on
+    """) with AnyWordSpecLike with LogCapturing {
+
+  implicit def executor: ExecutionContext = system.executionContext
+  override implicit val timeout: Timeout = 10.millis
+
+  def forwardBehavior: Behavior[Command] =
+    setup[Command] { context =>
+      Behaviors.receiveMessage[Command] { msg =>
+        msg match {
+          case Multiply(a, b, ref, replyTo) =>
+            // context.ask is asynchronous
+            context.ask[WorkerCommand, WorkerResult](ref, resultReply => WorkerMultiply(a, b, resultReply)) {
+              case Success(result) => ReplyResult(result.num, replyTo)
+              case Failure(_)      => Ignore()
+            }
+            Behaviors.same
+          case ReplyResult(num, replyTo) =>
+            replyTo ! num
+            Behaviors.same
+          case Ignore() => Behaviors.same
+        }
+      }
+    }
+
+  "DeadLetterActor" must {
+
+    "publish dead letter with recipient when context.ask terminated" in {
+      val forwardRef = spawn(forwardBehavior)
+      testDeadLetterPublishWhenAskTimeout[Int](ref => forwardRef.ask(replyTo => Multiply(3, 9, ref, replyTo)))
```

Comment:
To make it extra clear what is going on:
```suggestion
      testDeadLetterPublishWhenAskTimeout[Int](terminatedRef => 
        forwardRef.ask(replyTo => Multiply(3, 9, terminatedRef, replyTo)))
```
Same for the other test